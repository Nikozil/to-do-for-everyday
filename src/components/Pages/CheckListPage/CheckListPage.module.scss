@import '../../../App.scss';

.check-list {
  display: flex;
  justify-content: center;
  height: 100%;
}
.check-list__content {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 95%;
  width: 100%;
  min-width: 200px;
  max-width: 600px;
  margin: 0 5px 0 5px;
  text-align: center;

  font-size: 1.5em;
  color: $primary-color;
  @media screen and (max-width: 767px) {
    font-size: 1.2rem;
    margin: 0 15px 0 10px;
  }
}
.day-score {
  font-size: 0.8em;
}
.spinner {
  margin: 0 auto 0 auto;
  color: $primary-color !important;
}
