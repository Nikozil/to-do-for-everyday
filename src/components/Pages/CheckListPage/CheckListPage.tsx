import React from 'react';
import { useSelector } from 'react-redux';
import SpinComponent from '../../../common_components/SpinComponent/SpinComponent';
import { selectInitTasksStatus } from '../../../Redux/selectors/tasksSelector';
import NewTaskWithAutocompleteForm from '../../Forms/NewTaskWithAutocompleteForm/NewTaskWithAutocompleteForm';
import styles from './CheckListPage.module.scss';
import DayScoreComponent from './DayScoreComponent/DayScoreComponent';
import TasksListComponent from './TasksListComponent/TasksListComponent';

const CheckListPage = () => {
  const initTasksStatus = useSelector(selectInitTasksStatus);

  const duration = { days: 0 };

  return (
    <section className={styles['check-list']}>
      <div className={styles['check-list__content']}>
        <section className={styles.tasks}>
          <div className={'my-4'}>
            <NewTaskWithAutocompleteForm duration={duration} />
          </div>

          {initTasksStatus ? (
            <TasksListComponent />
          ) : (
            <SpinComponent styleClass={styles.spinner} />
          )}
        </section>

        <section className={styles['day-score']}>
          {initTasksStatus ? (
            <DayScoreComponent />
          ) : (
            <SpinComponent styleClass={styles.spinner} />
          )}
        </section>
      </div>
    </section>
  );
};

export default CheckListPage;
