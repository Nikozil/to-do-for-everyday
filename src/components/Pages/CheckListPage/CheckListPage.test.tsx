import React from 'react';
import { render } from '../../../utils/tests/test-utils';
import CheckListPage from './CheckListPage';

describe('CheckListPage', () => {
  it('render correct', async () => {
    const { asFragment } = render(<CheckListPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
