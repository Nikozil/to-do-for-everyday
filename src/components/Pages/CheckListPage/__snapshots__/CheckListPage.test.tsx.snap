// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckListPage render correct 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div
      class="content"
    >
      <div
        class="my-4"
      >
        <form
          action="#"
        >
          <div
            class="mb-1"
          >
            <input
              autocomplete="off"
              class="form-control newTask"
              id="newTask"
              name="newTask"
              placeholder="Новая Задача"
              type="text"
              value=""
            />
          </div>
        </form>
      </div>
      <div
        class="spinner-border  spinner"
        role="status"
      >
        <span
          class="visually-hidden"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
   
</DocumentFragment>
`;
